.clang-tidy

```diff
@@ -1,6 +1,6 @@
 Checks: "-*,\
 bugprone-*,\
-cert-dcl21-cpp,\
+-cert-dcl21-cpp,\
```

Comment:
Please re-enable. https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl21-cpp.html

Reply:
This was the result of a discussion between @pbukva and I. Lets discuss in the office, however, in brief, it looks like this check is not always that helpful in the case of post-increment operators.

Reply:
I'd be interested to hear a more detailed explanation. Also, last time I tried to define a postfix increment for an iterator, your take was that we should avoid them altogether. What changed?

Reply:
Ok, having spoken to @ejfitzgerald, let's roll with it for now. After this PR is in, I will try to re-enable this check and ininvestigate why it was failing.